src/HtmlUtils.js

```diff
@@ -208,17 +210,25 @@ module.exports = {
             finally {
                 delete sanitizeHtmlParams.textFilter;
             }
-            return <span className="markdown-body" dangerouslySetInnerHTML={{ __html: safeBody }} />;
         } else {
-            safeBody = content.body;
+            safeBody = _.escape(content.body);
```

Comment:
Firstly - sorry for having not managed to progress the review this week; I've been stuck doing mainly non-dev and the rest of the team has been elsewhere.  Meanwhile, this is a really security-sensitive part of Vector (in terms of the risk of XSS if the escaping and dangerouslySetInnerHTMLs aren't handled correctly, hence wanting to review it fully first).

So, previously, the way this works originally is that `TextHighlighter.applyHighlights()` returns a list of React `<span/>` nodes which wrap text nodes of the original plain text of the event body (which isn't HTML or escaped), as per the method's javascript doc.  React then automatically safely escapes these textnodes when rendering the spans to the DOM as normal, so they don't need to be escaped.  It's no different to doing `render() { foo='&amp;'; return <span>{ foo }</span>; }` - the `foo` variable gets automatically escaped and you see a literal `&amp;` string rendered by the browser.

To pose a different question to try to understand how this is meant to work: we seem to be taking the list of `<span/>`s returned by `TextHighlighter.applyHighlights()` and handing it straight to `emojione.toImage()`.  How does this handle this?  Is it somehow collapsing the React components back down to their raw string representations?  If so, this is a no-go because it will remove any highlights we just added.  On the other hand, if it's actually processing the React components correctly as components, they shouldn't need to be pre-escaped as the original logic should still work.  And one should not be handing them to dangerouslySetInnerHTML, which takes an escaped string, rather than a list of React components.

So... did you envisage this working? :)